private-eye.js

```diff
@@ -6,11 +6,36 @@
     styles.innerHTML = 'a.private-link::after { content: "\\1F512"; font-size: 0.75em; vertical-align: top; }';
     document.body.appendChild(styles);
 
+    var DEFAULT_MESSAGE = "This is a link to a private site, which may or may not be accessible to you.";
+    var hrefValue;
+    var linkMessage;
+
     opts.ignoreUrls.forEach(function(url) {
-      var anchors = document.querySelectorAll('a[href*="' + url + '"]');
+
+      hrefValue = url;
+      linkMessage = DEFAULT_MESSAGE;
+
+      // If the `url` is an Object, then parse the properties `message` & `url`
+      //
+      if ( url === Object( url ) ) {
+
+        linkMessage = url.message;
+        hrefValue = url.url;
+
+      }
+
+      var anchors = document.querySelectorAll('a[href*="' + hrefValue + '"]');
       Array.prototype.forEach.call(anchors, function(anchor) {
         anchor.className += ' private-link';
-        anchor.title = "This is a link to a private site, which may or may not be accessible to you.";
+
+        // Only replace the anchor's title if it is empty
+        //
+        if ( '' === anchor.title ) {
```

Comment:
Sure thing, better safe than sorry. Also, [fun fact: title attributes are inherited through their parents](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/title). Wonder if that is an issue here? Probably not.